core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -191,27 +195,35 @@ static void configure(Environment environment, boolean filterBadDefaults) throws
     @SuppressForbidden(reason = "accesses fully qualified URLs to configure security")
     static Policy readPolicy(URL policyFile, Set<URL> codebases) {
         try {
+            List<String> propertiesSet = new ArrayList<>();
             try {
                 // set codebase properties
                 for (URL url : codebases) {
                     String shortName = PathUtils.get(url.toURI()).getFileName().toString();
                     if (shortName.endsWith(".jar") == false) {
                         continue; // tests :(
                     }
-                    String previous = System.setProperty("codebase." + shortName, url.toString());
+                    if (shortName.startsWith("elasticsearch-rest-client")) {
+                        final String esVersion = Version.CURRENT + (Build.CURRENT.isSnapshot() ? "-SNAPSHOT" : "");
+                        System.setProperty("es.version", esVersion);
+                        propertiesSet.add("es.version");
+                        final String urlAsString = url.toString();
+                        final int index = urlAsString.indexOf("-" + System.getProperty("es.version") + ".jar");
+                        assert index >= 0;
+                        shortName = urlAsString.substring(0, index);
+                    }
+                    String property = "codebase." + shortName;
+                    propertiesSet.add(property);
+                    String previous = System.setProperty(property, url.toString());
```

Comment:
This doesn't work. Note that

```
"${codebase.elasticsearch-rest-client}-${es.version}.jar"
```

translates to

```
file:///path/to/elasticsearch-rest-client-7.0.0-alpha1-SNAPSHOT.jar-7.0.0-alpha1-SNAPSHOT.jar
```

because you're replacing `codebase.elasticsearch-rest-client` with the *full* URL string, and then appending the version (from `es.version`). I handled this in the [diff](https://github.com/elastic/elasticsearch/pull/26521#issuecomment-327618897) I gave.